source-mysql/main.go

```diff
@@ -223,7 +227,16 @@ func (db *mysqlDatabase) connect(ctx context.Context) error {
 		return fmt.Errorf("unable to connect to database: %w", err)
 	}
 
-	// Set our desired timezone to UTC. This is required for backfills of timestamp columns to behave consistently.
+	// Infer the location in which captured DATETIME values will be interpreted.
+	if loc, err := queryTimeZone(conn); err == nil {
+		db.datetimeLocation = loc
+		logrus.WithField("loc", loc.String()).Debug("using datetime location")
+	} else {
+		logrus.WithField("err", err).Warn("unable to determine database timezone")
+		logrus.Warn("capturing DATETIME values will not be permitted")
+	}
+
+	// Set our desired timezone to UTC. This is required for backfills of TIMESTAMP columns to behave consistently.
 	if _, err := db.conn.Execute("SET time_zone = '+00:00';"); err != nil {
```

Comment:
Making sure I'm following: we're fetching a persisted `time_zone` on startup, and then overwriting it _for our session only_ as part of making backfills work for TIMESTAMP? This is sound because DATETIME query results are not affected by setting `time_zone`, while TIMESTAMP will consistently be UTC, and no other sessions will observe our write to `time_zone`?

Reply:
That sounds exactly right. The read is of the global setting of `time_zone` which should apply to all sessions by default, and then we modify it for just our session so that the DB will do the right thing and give us `TIMESTAMP` values in UTC.

Reply:
I've tweaked the comment to hopefully clarify what's going on for future readers.